modules/compute/vm-instance/main.tf

```diff
@@ -61,6 +61,22 @@ locals {
   smt_capable          = local.smt_capable_family && local.smt_capable_vcpu && local.machine_not_shared_core
   set_threads_per_core = var.threads_per_core != null && (var.threads_per_core == 0 && local.smt_capable || try(var.threads_per_core >= 1, false))
   threads_per_core     = var.threads_per_core == 2 ? 2 : 1
+
+  # Network Interfaces
+  # Support for `use` input and base network paramters like `network_self_link` and `subnetwork_self_link`
+  base_network_interface = {
+    network            = var.network_self_link
+    subnetwork         = var.subnetwork_self_link
+    subnetwork_project = var.project_id
+    network_ip         = null
+    nic_type           = local.enable_gvnic ? "GVNIC" : null
+    stack_type         = null
+    queue_count        = null
+    access_config      = var.disable_public_ips ? [] : [{}]
+    ipv6_access_config = []
+    alias_ip_range     = []
+  }
+  network_interfaces = length(var.network_interfaces) == 0 ? [local.base_network_interface] : var.network_interfaces
```

Comment:
Nit: you might consider [coalescelist](https://www.terraform.io/language/functions/coalescelist). This function dates back to the 0.x days.

Reply:
Above a nit, below a show-stopper: I prefer "default" to "base" which suggests something that might always be there.

Reply:
coalescelist is much cleaner, thanks for the suggestion. Fixed both.